.github/workflows/custom_image_generation.yml

```
@@ -0,0 +1,159 @@
+# This will automatically use the sbt version in project/build.properties
+name: Custom Image Generation
+
+#TODO: should only be `workflow_dispatch` eventually, after testing
+on:
+#TODO: REMOVE BEFORE MERGE
+  push:
+    branches: 
+      - 'jc-image-action'
+  workflow_dispatch:
+    inputs:
+      run-dataproc-override:
+        description: 'Set this to `true` to force dataproc image generation script to run regardless of if there are changes to generation script.'
+        required: false
+        default: false
+        type: boolean
+      run-gce-override:
+        description: 'Set this to `true` to force gce image generation script to run regardless of if there are changes to generation script.'
+        required: false
+        default: false
+        type: boolean
+
+env:
+  GOOGLE_PROJECT: broad-dsp-gcr-public
+  GOOGLE_CREDENTIAL_FILE_NAME: image-build-account.json
+  REGION: us-central1
+  ZONE: us-central1-a
+  OUTPUT_BASE_NAME: custom-leo
+
+#   inspect-changes:
+#     runs-on: ubuntu-latest
+#     outputs:
+#       has-dataproc-changes: ${{ steps.filter.outputs.dataproc }}
+#       has-gce-changes: ${{ steps.filter.outputs.gce }}
+
+#     steps:
+#     - uses: actions/checkout@v3
+#     - uses: dorny/paths-filter@v2
+#       id: filter
+#       with:
+#         base: 'develop'
+#         filters: |
+#           dataproc:
+#             - 'jenkins/dataproc-custom-images/**' 
+#           gce:
+#             - 'jenkins/gce-custom-images/**' 
+
+#     - name: display progress
+#       run: | 
+#         echo "finished inspecting the repo to determine which image generation scripts to run@@@@@@@@@@@"
+
+jobs:
+  run-image-generation-script-gce:
+    runs-on: ubuntu-latest
+    # needs: inspect-changes
```

```suggestion
    permissions:
      contents: 'read'
      id-token: 'write'
    # needs: inspect-changes
```

You need to add this block in order for workload id federation to work